drivers/flash/nrf_qspi_nor.c

```diff
@@ -353,11 +350,13 @@ static int qspi_erase(struct device *dev, u32_t addr, u32_t size)
 			size -= adj;
 		} else {
 			LOG_ERR("erase error at 0x%lx size %zu", (long)addr, size);
-			return rv;
+			rv = qspi_get_zephyr_ret_code(res);
 		}
 	}
 
-	return 0;
+	qspi_unlock(dev);
```

Comment:
There will be double `qspi_unlock` when erase size is not sector aligned, because `res` will be set to `!NRFX_SUCCESS`,
and `qspi_wait_for_completion` will be invoked with that `res`, so it will go into `qspi_unlock` path.

Reply:
No: there is no `qspi_unlock()` path in `qspi_wait_for_completion()` anymore.

Reply:
Sorry got old codebase.